app/middleware/healthchecks/customHealthChecks.js

```diff
@@ -0,0 +1,72 @@
+const HealthCheck = require("@financial-times/health-check");
+const fetch = require('node-fetch');
+
+class AggregateHealthCheck extends HealthCheck.Check {
+    constructor(options) {
+        super(options);
+        this.url = options.url
+        this.graphiteKey = options.graphiteKey
+        this.timeSpan = options.timeSpan
+    }
+
+    async run() {
+        try {
+            // make call to both graphite endpoints
+            const response = await fetch(this.options.url, {
+                method: 'GET',
+                headers: { key: this.options.graphiteKey }
+            });
+
+            if (response.status !== 200) {
+                throw new Error('Non-200 response, check what is happening with the Graphite URL');
+            }
+
+            let data = await response.json();
+
+            // check if response array is null
+
+            if (data.length === 0) {
+                console.log('no fails: ' + this.options.url)
+                this.ok = true;
+                this.checkOutput = '';
+            }
+
+            // if not, reduce and check
+
+            else { 
+                let array = []
+
+                data.forEach((object) => {
+                    array.push(object.datapoints[0][0])
+                })
+
+                const sum = (accumulator, currentIndex) => {
+                    return (accumulator + currentIndex);
+                }
+
+                const total = array.reduce(sum)
+
```

Comment:
Tiny point but you could maybe make this a little more terse e.g.

```
const total = data.reduce((accumulator, currentValue) => {
  accumulator + currentValue.datapoints[0][0]
}, 0);
```

Reply:
Have updated it 👍